cms/djangoapps/contentstore/tests/test_course_settings.py

```diff
@@ -971,25 +973,35 @@ def test_advanced_components_munge_tabs(self):
         course = modulestore().get_course(self.course.id)
         self.assertNotIn(open_ended_tab, course.tabs)
         self.assertNotIn(peer_grading_tab, course.tabs)
-        self.assertNotIn(notes_tab, course.tabs)
-
-    def mark_wiki_as_hidden(self, tabs):
-        """ Mark the wiki tab as hidden. """
-        for tab in tabs:
-            if tab.type == 'wiki':
-                tab['is_hidden'] = True
-        return tabs
-
-    def test_advanced_components_munge_tabs_hidden_tabs(self):
-        updated_tabs = self.mark_wiki_as_hidden(self.course.tabs)
-        self.course.tabs = updated_tabs
+        self.assertNotIn(self.notes_tab, course.tabs)
+
+    def test_advanced_components_munge_tabs_validation_failure(self):
+        with patch('contentstore.views.course._refresh_course_tabs', side_effect=InvalidTabsException):
+            resp = self.client.ajax_post(self.course_setting_url, {
+                ADVANCED_COMPONENT_POLICY_KEY: {"value": ["notes"]}
+            })
+            self.assertEqual(resp.status_code, 400)
+
+            error_msg = [{'message': '', 'model': {'display_name': 'Tabs Exception'}}]
+            self.assertEqual(json.loads(resp.content), error_msg)
+
+            # verify that the course wasn't saved into the modulestore
+            course = modulestore().get_course(self.course.id)
+            self.assertNotIn("notes", course.advanced_modules)
+
+    @ddt.data(
```

Comment:
We do have a lot of tests around the tabs validation already: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/tests/test_tabs.py#L469

For these tests, I was trying to do more testing around the removing/adding of tabs that happens as part of Advanced Settings.